PrebidMobile/Rendering/PrebidMobileRendering/PrebidMobileRendering/Prebid/Integrations/PrimarySDKUtilHelpers/NativeAdDetectionListener.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  NativeAdDetectionListener.swift
+//  PrebidMobileRendering
+//
+//  Copyright © 2021 Prebid. All rights reserved.
+//
+
+import Foundation
+
+public typealias NativeAdLoadedHandler = (NativeAd) -> Void
+public typealias PrimaryAdServerWinHandler = () -> Void
+public typealias InvalidNativeAdHandler = (Error) -> Void
+
+/// Immutable container for 3 mutually exclusive outcomes of an asynchronous native ad detection attempt.
+public class NativeAdDetectionListener: NSObject, NSCopying {
+    
+    @objc public private(set) var onNativeAdLoaded: NativeAdLoadedHandler?
```

Comment:
These properties in the original file were marked as nonnull and  it makes sense since all of them are requred for this class. 

Can we make them non-optional here as well?

Reply:
They are marked as nullable:
```
@interface PBMNativeAdDetectionListener : NSObject <NSCopying>

@property (nonatomic, copy, nullable, readonly) PBMNativeAdLoadedHandler onNativeAdLoaded;
@property (nonatomic, copy, nullable, readonly) PBMPrimaryAdServerWinHandler onPrimaryAdWin;
@property (nonatomic, copy, nullable, readonly) PBMInvalidNativeAdHandler onNativeAdInvalid;
```